Reconstructive surgery for moth wings

Maurukas opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This adds a new advanced surgery which allows medbay to repair burned off moth wings with the use of synthflesh. It modifies the relevant tech node and medical designs to include this under the appropriate research. It also slightly modifies the mothmen species, as wings being burned off currently destroys that genetic information.

## Why It's Good For The Game

Currently burned off moth wings are the only mutant body part which cannot be repaired or reattached by medbay. In fact they cannot be restored by any in-game means (including magical healing and aheals) other than variable editing. This is something I have heard requested a number of times, and appears to be a fairly long standing issue.

Moth wings confer minor in game benefits and are part of a character's visual identity. At the price of time (and the cost of synthflesh) players now have the option of having an otherwise permanent injury fixed.

## Changelog
:cl:
add: Added reconstructive surgery for moth wings, requiring synthflesh
bugfix: Magical healing and aheals now restore moth wings
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Maurukas committed:
add wing reconstruction

Adds surgical reconstruction of moth wings.

Fikou commented:
what if a moth has roundstart burnt off wings? can you make it create default wings if it doesnt find any

Maurukas commented:
Currently in this PR if you pick the burnt wings in your preferences, it makes you ineligible for this surgery as /datum/species/moth/handle_fire won't apply to you (it checks to see if your wings are already burned). Reconstructing round start burnt wings to 'plain' should be pretty minor though.

Maurukas committed:
shift start burnt wings

Moths who enter NT employment with burnt wings may not take advantage of this surgery, receiving "Plain" wings.

Jared-Fogle commented:
Long term it would be cooler if moth wings were their own organs, so you could recreate this without research by getting a copy of yourself from genetics and transplanting the wings.

Jared-Fogle commented on code/modules/surgery/advanced/wingreconstruction.dm:
```diff
@@ -0,0 +1,40 @@
+	display_results(user, target, "<span class='notice'>You succeed in reconstructing [target]'s wings.</span>",
+		"<span class='notice'>[user] successfully reconstructs [target]'s wings!</span>",
+		"<span class='notice'>[user] completes the surgery on [target]'s wings.</span>")
+	if(H.dna.features["original_moth_wings"] != null)
+		H.dna.features["moth_wings"] = H.dna.features["original_moth_wings"]
+	if(H.dna.features["original_moth_wings"] == null)
+		H.dna.features["moth_wings"] = "Plain"
```
```suggestion
	H.dna.features["moth_wings"] = H.dna.features["original_moth_wings"] || "Plain"
```

Jared-Fogle replied:
Not sure if this is meant to be the "Burnt Off" check, but are you sure `original_moth_wings` isn't just "Burnt Off"?

Maurukas replied:
`original_moth_wings` should always be null, unless `handle_fire` inserts it into that list when it overwrites `moth_wings`. This proc won't do this if the wings already burnt off, which happens if you pick roundstart burnt wings.

Jared-Fogle replied:
The code suggestion still stands BTW, the question was just a followup.

Jared-Fogle commented on code/modules/surgery/advanced/wingreconstruction.dm:
```diff
@@ -0,0 +1,40 @@
+	target_mobtypes = list(/mob/living/carbon/human)
+
+/datum/surgery/advanced/wing_reconstruction/can_start(mob/user, mob/living/carbon/target)
+	if(!..())
+		return FALSE
+	if(target.dna.features["moth_wings"] == "Burnt Off")
+		return TRUE
```
```suggestion
	return ..() && target.dna.features["moth_wings"] == "Burnt Off"
```

ATH1909 commented:
good PR!

can you make staves of healing restore them too?

zombiegums commented:
Thank you for this. I always thought it odd that there was no way to fix tattered wings.

ExcessiveUseOfCobblestone commented on code/modules/surgery/advanced/wingreconstruction.dm:
```diff
@@ -0,0 +1,40 @@
+		"<span class='notice'>[user] completes the surgery on [target]'s wings.</span>")
+	if(H.dna.features["original_moth_wings"] != null)
+		H.dna.features["moth_wings"] = H.dna.features["original_moth_wings"]
+	if(H.dna.features["original_moth_wings"] == null)
```
`else` instead?

If the first check didnt go through we know the second one must.

Jared-Fogle replied:
I made a comment that's less cluttered than an if condition above if you didn't see.

Maurukas replied:
Accepted Jared-Fogle's suggestion above.

ExcessiveUseOfCobblestone commented on code/modules/surgery/advanced/wingreconstruction.dm:
```diff
@@ -0,0 +1,40 @@
+		"<span class='notice'>[user] begins to perform surgery on [target]'s charred wing membranes.</span>")
+
+/datum/surgery_step/wing_reconstruction/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	var/mob/living/carbon/human/H = target
```
assuming!

Maurukas replied:
Pardon, not quite following. Do I need a check for ishuman/ismoth here?

Maurukas replied:
Dealt with. Please be patient I'm special.

Jack7D1 commented:
Speedmerge

RaveRadbury commented:
Wasn't this a deliberate design choice?

Maurukas committed:
Add Jared's suggestion

Co-authored-by: Jared-Fogle <35135081+Jared-Fogle@users.noreply.github.com>

Maurukas committed:
aheals, suggested logic cleanup

Maurukas committed:
unassuming

Maurukas commented:
I haven't seen anything that indicates this was a design decision, it just looks like an artifact of how they were implemented. Aheals not working was definitely not intended - it's been fixed before.

Maurukas committed:
resolve conflicts with dissection removal
